### ISSUE TITLE
Exceeding program memory on the arduino uno

### ISSUE THREAD

**Issue by Oriebugg** [slviajero/tinybasic#69]
Is the Arduino Uno r3 still supported ? I've tried flashing basic1 onto my Uno but the IDE says it exceeds the program memory by 119%

**Comment by slviajero** [comment]
yes supported which language and hardware settings do you use?Sent from my iPhoneOn Oct 29, 2024, at 6:30 PM, Oriebugg ***@***.***> wrote:﻿
Is the Arduino Uno r3 still supported ? I've tried flashing basic1 onto my Uno but the IDE says it exceeds the program memory by 119%

**Comment by slviajero** [comment]
Hi, just a few hints here. In version 1 all language settings are at the beginning of the .ino file. 

#undef  BASICFULL
#undef  BASICINTEGER
#define BASICSIMPLE
#undef  BASICMINIMAL
#undef  BASICSIMPLEWITHFLOAT
#undef  BASICTINYWITHFLOAT


An UNO runs with BASICSIMPLE, which is an integer BASIC. 

In addition to this setting, I recommend USEPICOSERIAL in hardware-arduino.h.

I corrected the default settings in the repository.

**Comment by Oriebugg** [comment]
Hi! Thank you so much for your reply. I got it working now!! If I may ask, what's picoserial?

**Comment by slviajero** [comment]
Sorry, did not see the last question. PICOSERIAL is a serial code which does not use the Serial object of the UNO. Much leaner. Not portable to other platforms.
